### DISCUSSION TITLE
How to create a plugin which adds global features?

### DISCUSSION THREAD

**Discussion by Kunsi** [pretalx/pretalx#1203] [Q&A] [answered]
Hi,

i'm trying to implement https://github.com/seibert-media/pretalx-redirector as a pretalx plugin. However all plugin documentation seems to be for creating a *event plugin*, which isn't available globally.

Am i missing something? Is this the correct approach, or would creating redirects `<pretalx url>/<talk id>` need to be a core feature?

**Comment by rixx** [comment, marked as answer]
Maybe I'm missing something, but I don't see what event plugin signals you would need for this to work. Wouldn't you just add a single URL and view, and be done with the plugin?

*Edit:* https://github.com/rixx/pretalx-proposal-redirects

**Reply by rixx** [reply]
That said, proposal codes are probably going to stop being globally unique, and are going to move to being only unique within their event, so this redirector pattern isn't going to be working indefinitely.

**Reply by Kunsi** [reply]
Oh, that's sad to hear. May i ask why that change will happen?

**Reply by rixx** [reply]
We might get away with increasing the default length from 6 to 7, tbh, which is why I have not decided yet.

Does the plugin work? Happy to make it more available / add you as collaborator. It's still missing checks for if the selected proposal is public (or if the requesting user has access), so it's currently leaking the existence of non-public proposals if you guess their ID, but other than that, it should be good to go.

**Reply by Kunsi** [reply]
I have not yet tried. I will test over the weekend.

**Reply by fkusei** [reply]
Just tested it, works fine. Will file some PRs against it, to add the mentioned safeguards, also i'd like to add `/<slug>/feedback` or something to directly link to the feedback page.
